### PR TITLE
SASS-2174 - Create CIS materials amount EOY page.

### PR DESCRIPTION
**Pull request by georgievh**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-2174](https://jira.tools.tax.service.gov.uk/browse/SASS-2174)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

**Comment by lz1kwn** [comment]
bootstrap-frontend-play-28 is at 5.23.0

**Comment by lz1kwn** [comment]
hmrc-mongo-play-28 is at 0.63.0

**Comment by georgievh** [comment]
uk.gov.hmrc.mongo: hmrc-mongo-play-28 is up to date. Version 0.63.0 is already the latest, isn't it?

**Comment by georgievh** [comment]
bootstrap-frontend-play-28 cannot be updated beyond 5.20.0 at this point

**Comment by lz1kwn** [comment]
bootstrap-frontend-play-28 cannot be updated beyond 5.20.0 at this point, thank you
uk.gov.hmrc.mongo: hmrc-mongo-play-28 is up to date. Version 0.63.0 is already the latest, isn't it?, it is showing o.62.0 in IntelliJ

**Comment by georgievh** [comment]
I think you are looking at the wrong branch. The correct branch is: SASS-2174-hristo

**Review by lz1kwn — CHANGES_REQUESTED** [review]
if the answer to 'paid for materials' radio button is 'no' then the user shouldn't be able to access this amount page, this is no the case at the moment and needs to be completed

**Comment by georgievh** [comment]
Liz, are you sure you are running the right branch? Your comment about selecting no on the radio button works correctly. There are even tests for that.

File: conf/messages [hunk 1]
```diff
@@ -158,6 +158,27 @@ materialsPage.warning = This does not include work related expenses (fuel, hotel
 materialsPage.error.agent = Select yes if your client paid for materials
 materialsPage.error.individual = Select yes if you paid for materials
 
+#################   Materials Amount Page   ############################
+
+materialsAmountPage.title.individual = How much did you pay for building materials for your contractor?
+materialsAmountPage.title.agent = How much did your client pay for building materials for the contractor?
+materialsAmountPage.heading.individual = How much did you pay for building materials at {0}?
+materialsAmountPage.heading.agent = How much did your client pay for building materials at {0}?
+materialsAmountPage.replayContent.p1.individual = If you did not pay {0} for materials,  tell us the correct amount.
```
Comment (ina-tu) [inline conf/messages:167]: there's an extra space after the comma for `materialsAmountPage.replayContent.p1.individual`. This is for both the welsh and english message

Comment (georgievh) [reply conf/messages]: Fixed.

File: test/views/MaterialsAmountViewSpec.scala [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+        welshToggleCheck(userScenario.isWelsh)
+        titleCheck(userScenario.specificExpectedResults.get.expectedErrorTitle)
+        captionCheck(userScenario.commonExpectedResults.expectedCaption(taxYearEOY))
+        h1Check(userScenario.specificExpectedResults.get.expectedH1("some-contractor"))
+        textOnPageCheck(userScenario.specificExpectedResults.get.expectedTellUsTheAmountText(translatedMonthAndTaxYear(pageModel.month, taxYearEOY)), Selectors.paragraphTextSelector(number = 2))
+        textOnPageCheck(userScenario.specificExpectedResults.get.expectedOnlyIncludeVATParagraph, Selectors.paragraphTextSelector(number = 3))
+        hintTextCheck(userScenario.commonExpectedResults.expectedHintText, Selectors.hintTextSelector)
+        textOnPageCheck(text = "£", Selectors.poundPrefixSelector)
+        errorSummaryCheck(userScenario.specificExpectedResults.get.expectedErrorNoEntry, Selectors.expectedErrorHref)
```
Comment (ina-tu) [inline test/views/MaterialsAmountViewSpec.scala:198]: we're missing the check for the error message that appears above the amount field box (errorAboveElementCheck). It looks like this has been missed in our other view specs too. Is this something we want to add now or in a follow-up/refactoring PR?

Comment (ina-tu) [reply test/views/MaterialsAmountViewSpec.scala:198]: as discussed this is part of a different element and does not need to be tested here

**Review by ina-tu — APPROVED** [review]

**Review by lz1kwn — APPROVED** [review]

**Pull request merged**